### PR TITLE
[feature] Support retrieving list of clients

### LINKED ISSUE #45: Add support for listing registered clients.
Currently the `org.osiam.auth.oauth_client.ClientManagementController` doesn't support to list all registered clients. It would be useful to have such a listing as a foundation for additional admin tools.

### PR DESCRIPTION
**Pull request by dacrome**
Closes #45

### CODE REVIEW

File: src/main/java/org/osiam/auth/oauth_client/ClientDao.java [hunk 1]
```diff
@@ -42,6 +42,11 @@ public ClientEntity getClient(final String id) {
         return getClientById(id);
     }
 
+    public List getClients() {
```
Comment (tkrille) [inline src/main/java/org/osiam/auth/oauth_client/ClientDao.java]: Shouldn't that be `List<ClientEntity>`?

File: src/main/java/org/osiam/auth/oauth_client/ClientManagementController.java [hunk 2]
```diff
@@ -58,6 +59,12 @@ public ClientEntity getClient(@PathVariable final String id) {
         return clientDao.getClient(id);
     }
 
+    @RequestMapping(method = RequestMethod.GET)
+    @ResponseBody
+    public List getClients() {
```
Comment (tkrille) [inline src/main/java/org/osiam/auth/oauth_client/ClientManagementController.java]: Shouldn't that be `List<ClientEntity>`?

File: src/main/java/org/osiam/auth/oauth_client/ClientDao.java (hunk 1 above)
Comment (dacrome) [inline src/main/java/org/osiam/auth/oauth_client/ClientDao.java]: Yes.

File: src/main/java/org/osiam/auth/oauth_client/ClientManagementController.java (hunk 2 above)
Comment (dacrome) [inline src/main/java/org/osiam/auth/oauth_client/ClientManagementController.java]: Yes.

**Comment by dacrome** [comment]
updated

File: src/main/java/org/osiam/auth/oauth_client/ClientEntity.java [hunk 3]
```diff
@@ -39,7 +39,10 @@
 @JsonSerialize(include = JsonSerialize.Inclusion.NON_EMPTY)
 @Entity
 @Table(name = "osiam_client")
-@NamedQueries({ @NamedQuery(name = "getClientById", query = "SELECT i FROM ClientEntity i WHERE i.id= :id") })
+@NamedQueries({
+        @NamedQuery(name = "getClientById", query = "SELECT i FROM ClientEntity i WHERE i.id= :id"),
+        @NamedQuery(name = "getClients", query = "SELECT i FROM ClientEntity i")
```
Comment (tkrille) [inline src/main/java/org/osiam/auth/oauth_client/ClientEntity.java]: Why `i`?

**Comment by dacrome** [comment]
Updated.

**Comment by dacrome** [comment]
Corresponding integration-tests https://github.com/osiam/connector4java-integration-tests/pull/262

File: src/main/java/org/osiam/auth/oauth_client/ClientDao.java [hunk 4]
```diff
@@ -42,6 +42,11 @@ public ClientEntity getClient(final String id) {
         return getClientById(id);
     }
 
+    public List<ClientEntity> getClients() {
+        final Query query = em.createNamedQuery("getClients");
```
Comment (tkrille) [inline src/main/java/org/osiam/auth/oauth_client/ClientDao.java]: Please use `javax.persistence.EntityManager#createNamedQuery(java.lang.String, java.lang.Class<T>)`. This way you get a `TypedQuery` and there will be no hassle with the unchecked assignment in the line below.

**Comment by dacrome** [comment]
updated

**Pull request merged**